lib/spack/spack/test/llnl/util/tty/log.py

```diff
@@ -62,6 +62,23 @@ def test_log_python_output_without_echo(capfd, tmpdir):
         assert capfd.readouterr()[0] == ''
 
 
+@pytest.mark.skipif(sys.version_info.major == 2 or sys.version_info.minor == 5,
```

Comment:
```suggestion
@pytest.mark.skipif(sys.version_info[:2] == (3, 5),
```

I agree your guard works and likely there's not gonna be a pyton 4.5, but ... you never know :p